src/attackmate/variablestore.py

```diff
@@ -1,19 +1,71 @@
 from string import Template
+import re
 from typing import Any, Optional
 
 
+class ListParseException(Exception):
+    """ Exception for all List-Parser
+
+    This exception is raised by parse_list if anything
+    goes wrong.
+    """
+    pass
+
+
+class VariableNotFound(Exception):
+    """ Exception for all List-Parser
+
+    This exception is raised by get_variable if the
+    variable does not exist in the variablestore.
+    """
+    pass
+
+
+class ListTemplate(Template):
+    idpattern = r'(?a:[\[\]_a-z][\[\]_a-z0-9]*)'
+
+
 class VariableStore:
     def __init__(self):
         self.clear()
 
     def clear(self):
+        self.lists: dict[str, list[str]] = {}
         self.variables = {}
 
+    @classmethod
+    def is_list(cls, variable: str) -> bool:
+        if re.search(r'\[[0-9]+\]\Z', variable):
+            return True
+        else:
+            return False
+
+    @classmethod
+    def parse_list(cls, variable: str) -> tuple[str, int]:
+        parsed = re.search(r'\A([^\[\]]+)\[([0-9]+)\]\Z', variable)
+
+        if parsed is None:
+            raise ListParseException('List could not be parsed')
+
+        if parsed.group(2) is None:
+            raise ListParseException('List-value is None')
+        else:
+            value = parsed.group(2)
+
+        return (parsed.group(1), int(value))
+
+    def get_lists_variables(self) -> dict[str, str]:
```

Comment:
to make it easier to understand what happens here, variable naming:
```
def get_lists_variables(self) -> dict[str, str]:
    all_indexed_list_values = {}
    for list_name, list in self.lists.items():
        for index, value in enumerate(list):
            all_indexed_list_values[f'{list_name}[{index}]'] = value
    return all_indexed_list_values
```

hope i got everything right, all_indexed_list_values maybe still a bit clunky